apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.YCSB/Program.cs

```diff
@@ -0,0 +1,513 @@
+            }
+            else
+            {
+                //target worker version
+                RunWithTargetWorkers(GetOption<int>(s_num_worker)).WaitWithUnwrappedExceptions();
+            }
+            PrintResults(overallTime.ElapsedMilliseconds);
+
+            return 0;
+        }
+
+        /// <summary>
+        /// Runs a single worker task
+        /// </summary>
+        private static async Task PreWarm()
+        {
+            using (var spannerConnection = new SpannerConnection(
+                $"Data Source=projects/{s_project}/instances/{s_instance}/databases/{s_database}"))
+            {
+                await spannerConnection.OpenAsync();
+            }
+        }
+
+        /// <summary>
+        /// Runs a single worker task that targets a specific QPS without regard to workers.
+        /// (this is from C# spanner stress tests)
+        /// </summary>
+        private static async Task RunWorkerWithQps(double queriesPerSecond, TimeSpan testTime)
+        {
+            var sw = Stopwatch.StartNew();
+            var all = new List<Task>();
+
+            while (sw.Elapsed < testTime)
+            {
+                if (sw.Elapsed.TotalSeconds * queriesPerSecond > all.Count)
+                {
+                    var operationTarget = s_rand.Value.NextDouble() * s_total;
+                    var operation = 0;
+                    operationTarget -= s_operations[operation].Proportion;
+                    while (operationTarget > 0 && operation < s_operations.Count)
+                    {
+                        operation++;
+                        if (operation < s_operations.Count)
+                        {
+                            operationTarget -= s_operations[operation].Proportion;
+                        }
+                    }
+                    try
+                    {
+                        all.Add(s_operations[Math.Min(operation, s_operations.Count - 1)].Action(
+                            s_operations[Math.Min(operation, s_operations.Count - 1)].Latencies));
+                    }
+                    catch (Exception e)
+                    {
+                        DebugMessage($"This error is being ignored and we are continuing the test: {e.Message}");
+                    }
+                }
+            }
+
+            var timeout = Task.Delay(TimeSpan.FromSeconds(10));
+            await Task.WhenAny(Task.WhenAll(all), timeout);
+            s_totaloperations = all.Count;
+        }
+
+        /// <summary>
+        /// Runs a single worker task
+        /// </summary>
+        /// <param name="numWorkers"></param>
+        private static async Task RunWithTargetWorkers(int numWorkers)
+        {
+            //yield to let Main continue through.
+            await Task.Yield();
+            var parallelWorkers = new List<Task>();
+            var operation = ChooseOperation();
+
+            while (Interlocked.Decrement(ref s_operationsRemaining) > 0)
```

Comment:
I don't understand why this is a global variable at all, rather than just a simple for loop with a local variable up to `s_totalOperations`.

Reply:
in a previous iteration, I had several worker tasks, so this needed to be shared.  it now doesnt even need the interlocked.